packages/protocol/contracts/L1/libs/LibProposing.sol

```diff
@@ -94,9 +95,15 @@ library LibProposing {
 
             if (blobHash == 0) revert L1_NO_BLOB_FOUND();
         } else {
+            // The proposer must be an Externally Owned Account (EOA) for
+            // calldata usage. This ensures that the transaction is not an
+            // internal one, making calldata retrieval more straightforward.
+            if (!msg.sender.isEOA()) revert L1_PROPOSER_NOT_EOA();
```

Comment:
This one: https://github.com/taikoxyz/taiko-mono/pull/15061#issuecomment-1788574549